ops/model.py

```diff
@@ -1009,9 +1015,15 @@ def relation_set(self, relation_id, key, value, is_app):
         if not isinstance(is_app, bool):
             raise TypeError('is_app parameter to relation_set must be a boolean')
 
+        if is_app and not JujuVersion.from_environ().has_app_data():
+            raise RuntimeError('trying to relation-set app data on a Juju too old for it')
```

Comment:
we'd want similar wording here.
I we are exposing this as 'relation-set' then I would probably say 'relation-set --app'.
If we are trying to talk it terms of Python objects, then I would say something more like:
'setting application relation data is not supported on older Juju versions'
or 'on Juju {version}'
or 'on Juju versions older than 2.7.0'

Reply:
I like the second one :-)
maybe `[SG]etting application data is not supported on Juju versions older than 2.7.0 (you have {})`

Reply:
hm, actually mentioning 2.7.0 in the error means we track it in two different places